buffer/src/main/java/io/netty/buffer/api/bytebuffer/NioBuffer.java

```diff
@@ -62,14 +62,15 @@
      * Constructor for {@linkplain BufferAllocator#constBufferSupplier(byte[]) const buffers}.
      */
     NioBuffer(NioBuffer parent) {
-        super(new ArcDrop<>(ArcDrop.acquire(parent.unsafeGetDrop())), parent.control);
+        super(parent.unsafeGetDrop().fork(), parent.control);
         base = parent.base;
         rmem = bbslice(parent.rmem, 0, parent.rmem.capacity()); // Need to slice to get independent byte orders.
         assert parent.wmem == CLOSED_BUFFER;
         wmem = CLOSED_BUFFER;
         roff = parent.roff;
         woff = parent.woff;
         constBuffer = true;
+        unsafeGetDrop().attach(this);
```

Comment:
calling `this` from within a constructor is kind of a bad-pattern as it may be possible to pass down a object which is not fully constructed yet. Consider using a static method that will first create the `NioBuffer` and then attach it.